src/components/tooltip/Tooltip.tsx

```diff
@@ -195,12 +197,13 @@ class Tooltip extends React.Component<TooltipProps, State> {
     };
 
     isShown = () => {
-        const { isShown: isShownProp } = this.props;
+        const { isShown: isShownProp, shouldTooltipShowAferUserClose = false } = this.props;
```

Comment:
Am wondering if you can implement a `componentDidUpdate` and watch for `isShown` changes. If it changes from false to true, set `wasClosedByUser` to false. Or, if you can somehow render the tooltip with a key which would force a re-mount. Your solution also works, just trying to avoid adding a new optional prop

Reply:
will play around with this to see if i can get it to work without the prop